### ISSUE TITLE
Suggestion of Phasor representation and object

### ISSUE THREAD

**Issue by Barleysack** [pytorch/pytorch#65337]
## 🚀 Feature
===================================
### I'd Like to suggest 'phasor' 👼 

[For the people who need concept of phasor](https://en.wikipedia.org/wiki/Phasor)

Which can simplify the usage of torch.complex during computations, making it possible to represent the continuous 
time-series data little bit more intuitively and get some advantage during computations.

## Motivation

Well..I'm just a noob NLP dev, trying to study torch little deeper.
Above that, I majored in electrical engineering, which contains quite a lot of calculations in frequency domain for time-series electric power data which also can be represented in angular domain and complex numbers. 

One day, I just got a curiosity from the fact that we are not utilizing the angular domain a lot in torch even if we are 
using tons of oscillating time- domain data. I already searched similar issues, but I could not find one...We all know there's torch.complex , which makes it possible to compute things in complex space. However, it is surprisingly not used often except few papers. (I was quite admired with A Survey of Complex-Valued Neural Networks,Joshua Bassey, Jan 2021) 

I believe phasor can make the complex computations easier since it can simplify some operations between complex numbers. For example, Multiplication and division between complex numbers can be changed into simple addition and subtraction of phasor data and simple abs value operations. 

That's why I'm writing this issue, and I just want to share the idea about it. 

## Pitch

what I want to suggest is to make phasor function which uses torch.complex object as input, making the phasor object output which only contains abs magnitude of complex number and angle of the vector representing the input.  
When we multiply between two phasor object, we will just multiply the magnitude element, and add the angle data to make output. Operation itself is quite simple. 

## Alternatives

Actually, library of cmath already have phasors. I think we can use it if we only want to calculate the simple value of phasor. 
However, I want to make the phasor object and simplify the procedures to calculate the magnitude and angle of complex number. Also, I hope we can make it sure if it will support training sequence of torch and other functions...I definitely want to utilize the complex numbers into neural network easier.

## Additional context

[Paper about complex neural network](https://arxiv.org/pdf/2101.12249.pdf)
[Documentation page of phasor in cmath library](https://docs.python.org/3/library/cmath.html)

I'm waiting for your kind suggestions and discussion about this idea :) 😄 

cc @ezyang @anjali411 @dylanbespalko @mruberry @Lezcano @nikitaved

**Comment by ezyang** [comment]
I am not an expert, but are phasors the same as polar complex numbers? https://github.com/pytorch/pytorch/issues/36175

**Comment by Barleysack** [comment]
> I am not an expert, but are phasors the same as polar complex numbers? #36175
Oh. yup.  Using polar form to represent something's amplitude and phase is phasor expression. 
I just searched for phasor and angle methods and I couldn't find one so I wrote this issue but it seems like we already had one :) 
I'll close this issue :) Thanks for the help!
